src/pdb/mod.rs

```diff
@@ -718,6 +720,101 @@ pub struct Track {
+        self.file_size.write_options(writer, options, ())?;
+        self.unknown2.write_options(writer, options, ())?;
+        self.unknown3.write_options(writer, options, ())?;
+        self.unknown4.write_options(writer, options, ())?;
+        self.artwork_id.write_options(writer, options, ())?;
+        self.key_id.write_options(writer, options, ())?;
+        self.orig_artist_id.write_options(writer, options, ())?;
+        self.label_id.write_options(writer, options, ())?;
+        self.remixer_id.write_options(writer, options, ())?;
+        self.bitrate.write_options(writer, options, ())?;
+        self.track_number.write_options(writer, options, ())?;
+        self.tempo.write_options(writer, options, ())?;
+        self.genre_id.write_options(writer, options, ())?;
+        self.album_id.write_options(writer, options, ())?;
+        self.artist_id.write_options(writer, options, ())?;
+        self.id.write_options(writer, options, ())?;
+        self.disc_number.write_options(writer, options, ())?;
+        self.play_count.write_options(writer, options, ())?;
+        self.year.write_options(writer, options, ())?;
+        self.sample_depth.write_options(writer, options, ())?;
+        self.duration.write_options(writer, options, ())?;
+        self.unknown5.write_options(writer, options, ())?;
+        self.color.write_options(writer, options, ())?;
+        self.rating.write_options(writer, options, ())?;
+        self.unknown6.write_options(writer, options, ())?;
+        self.unknown7.write_options(writer, options, ())?;
+
+        let start_of_string_section = writer.stream_position()?;
+        debug_assert_eq!(start_of_string_section - base_position, 0x5e);
+
+        // Skip offsets, because we want to write the actual strings first.
+        let mut string_offsets = [0u16; 21];
+        writer.seek(SeekFrom::Current(0x2a))?;
+        for (i, string) in [
+            &self.isrc,
+            &self.unknown_string1,
+            &self.unknown_string2,
+            &self.unknown_string3,
+            &self.unknown_string4,
+            &self.message,
+            &self.kuvo_public,
+            &self.autoload_hotcues,
+            &self.unknown_string5,
+            &self.unknown_string6,
+            &self.date_added,
+            &self.release_date,
+            &self.mix_name,
+            &self.unknown_string7,
+            &self.analyze_path,
+            &self.analyze_date,
+            &self.comment,
+            &self.title,
+            &self.unknown_string8,
+            &self.filename,
+            &self.file_path,
+        ]
+        .into_iter()
+        .enumerate()
+        {
+            let current_position = writer.stream_position()?;
+            let offset: u16 = current_position
+                .checked_sub(base_position)
+                .and_then(|v| u16::try_from(v).ok())
+                .ok_or_else(|| binrw::Error::AssertFail {
+                    pos: 0,
```

Comment:
this makes sense to do imo
```suggestion
                    pos: current_position,
```